hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -145,16 +235,29 @@ public RocksDBCheckpointDiffer(String dbPath,
     //  mapping.
   }
 
+  /**
+   * Helper function that recursively deletes the dir. TODO: REMOVE
+   */
+  boolean deleteDirectory(File directoryToBeDeleted) {
+    File[] allContents = directoryToBeDeleted.listFiles();
+    if (allContents != null) {
+      for (File file : allContents) {
+        deleteDirectory(file);
+      }
+    }
+    return directoryToBeDeleted.delete();
+  }
+
   // Node in the DAG to represent an SST file
   private class CompactionNode {
-    public String fileName;   // Name of the SST file
-    public String snapshotId; // The last snapshot that was created before this
+    private String fileName;   // Name of the SST file
+    private String snapshotId; // The last snapshot that was created before this
     // node came into existance;
-    public long snapshotGeneration;
-    public long totalNumberOfKeys;
-    public long cumulativeKeysReverseTraversal;
+    private long snapshotGeneration;
+    private long totalNumberOfKeys;
+    private long cumulativeKeysReverseTraversal;
 
-    CompactionNode (String f, String sid, long numKeys, long compactionGen) {
+    CompactionNode(String f, String sid, long numKeys, long compactionGen) {
```

Comment:
compactionGen parameter isn't used

Reply:
Yes. In the latest commit it has been repurposed to pass in the sequence number as snapshot generation.